jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/grid/TreeDataGrid.java

```diff
@@ -421,4 +425,36 @@ protected void onAfterApplyColumnSecurity(AbstractGridDelegate.ColumnSecurityCon
             super.removeColumn(context.getColumn());
         }
     }
+
+    @Override
+    public JmixGridContextMenu<E> getContextMenu() {
+        if (contextMenu == null) {
+            contextMenu = new JmixGridContextMenu<>();
+            contextMenu.setTarget(this);
+        }
+        return contextMenu;
+    }
+
+    @Override
+    public GridContextMenu<E> addContextMenu() {
+        throw new UnsupportedOperationException("TreeDataGrid can have only one context menu attached, " +
+                "use getContextMenu() to retrieve it");
+    }
+
+    @Nullable
+    @Override
+    public Object getSubPart(String name) {
+        Object column = super.getSubPart(name);
+        if (column != null) {
```

Comment:
Can it be moved to GridDelegate?

Reply:
No, because it uses `contextMenu` field:
```
    @Nullable
    @Override
    public Object getSubPart(String name) {
        Object column = super.getSubPart(name);
        if (column != null) {
            return column;
        }
        if (contextMenu != null) {
            if (UiComponentUtils.sameId(contextMenu, name)) {
                return contextMenu;
            } else {
                return contextMenu.getSubPart(name);
            }
        }
        return null;
    }
```
 The delegate doesn't have access to it (`getContextMenu()` has side effect - creates menu if not exist)